va/http.go

```diff
@@ -505,9 +503,12 @@ func (va *ValidationAuthorityImpl) processHTTPValidation(
 		numRedirects++
 		va.metrics.http01Redirects.Inc()
 
-		// If the response contains an HTTP 303 redirect, do not follow.
-		if req.Response.StatusCode == 303 {
-			return berrors.ConnectionFailureError("Cannot follow HTTP 303 redirects")
+		// If the response contains an HTTP 303 or any other forbidden redirect, do not follow.
```

Comment:
I think it's worth noting in the comment the reason why we're making this more explicit. Otherwise, someone down the line may look at this and say, "oh, the stdlib just does this", and revert this change.